packages/@aws-cdk-testing/framework-integ/test/aws-cloudfront-origins/test/integ.function-url-origin-oac.ts

```diff
@@ -0,0 +1,82 @@
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
+import * as cdk from 'aws-cdk-lib';
+import { ExpectedResult, IntegTest, Match } from '@aws-cdk/integ-tests-alpha';
+import * as origins from 'aws-cdk-lib/aws-cloudfront-origins';
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'integ-cloudfront-function-url-origin-oac');
+
+// Create the Lambda function with a custom response
+const fn = new lambda.Function(stack, 'MyFunction', {
+  code: lambda.Code.fromInline(`
+    exports.handler = async () => ({
+      statusCode: 200,
+      body: 'Hello!!'
+    });
+  `),
+  handler: 'index.handler',
+  runtime: lambda.Runtime.NODEJS_20_X,
+});
+
+// Add a Lambda Function URL
+const fnUrl = fn.addFunctionUrl({
+  authType: lambda.FunctionUrlAuthType.AWS_IAM,
```

Comment:
Just for my understanding, I would like to understand the behaviour of `AuthType` in Function URL with Cloudfront-Lambda integration with OAC origin.

1. May i know if `authType = AWS_IAM` is the only supported lamda function URL for Cloudfront OAC Lambda Function url origin?
2. Or Is `authType = None` supported for OAC? If it is not supported, Let us mention it in ReadMe and also update the current example.
3. If both are supported, I would recommend to have unit test for both types.

Reply:
OAC can communicate when authType = none and the necessary permissions are granted. When None is set in the L2 construct, the latter permission is automatically configured, so there are no issues. Therefore, I have prepared a test for the None case.

Reply:
I agree with @godwingrs22 . It's appropriate to throw an exception if `authType` is `None` and `OAC` Signing is `SIGV4_ALWAYS`. This configuration leads to errors and can be confusing for the customer.